File: python/06_021.py
def dfs_three_ways(node):
    if not node:
        return
        
    # Pre-order: Process BEFORE children
    print(node.value)  # Pre-order
    dfs_three_ways(node.left)
    # In-order: Process BETWEEN children
    print(node.value)  # In-order
    dfs_three_ways(node.right)
    # Post-order: Process AFTER children
    print(node.value)  # Post-order

File: cpp/06_021.cpp
void dfsThreeWays(TreeNode* node) {
    // Base case: return if node is null
    if (node == nullptr) {
        return;
    }
    
    // Pre-order: Process BEFORE children
    std::cout << node->value << std::endl;  // Pre-order
    dfsThreeWays(node->left);
    // In-order: Process BETWEEN children
    std::cout << node->value << std::endl;  // In-order
    dfsThreeWays(node->right);
    // Post-order: Process AFTER children
    std::cout << node->value << std::endl;  // Post-order
}